rpc/dataconn/timeoutconn/timeoutconn.go

```diff
@@ -73,9 +73,9 @@ func (c *Conn) DisableTimeouts() error {
 	c.mu.Lock()
 	defer c.mu.Unlock()
 	if !c.renewDeadlinesDisabled {
+		c.renewDeadlinesDisabled = true
```

Comment:
Hm, shouldn't we do this only if the c.SetDeadline call below doesn't return an error?

Reply:
I agree, that makes the most sense 👍🏻